packages/sado-connect/src/components/SelectWalletModal/index.tsx

```diff
@@ -59,25 +71,39 @@ export function SelectWalletModal({
       console.error("Error while connecting to UniSat wallet", err);
     }
   };
-  // const onConnectXverseWallet = async () => {
-  //   try {
-  //     const xverse = await ordit.xverse.getAddresses({
-  //       network,
-  //     });
-  //     const xverseWallet = xverse[0];
-  //     updateAddress(xverseWallet.address);
-  //     updatePublicKey(xverseWallet.pub);
-  //     updateWallet(Wallet.XVERSE);
-  //     updateFormat(xverseWallet.format as AddressFormats);
-  //     closeModal();
-  //   } catch (err: any) {
-  //     if (err?.message === "Xverse not installed.") {
-  //       window.open(XVERSE_WALLET_CHROME_EXTENSION_URL);
-  //     }
-  //     setErrorMessage(err.toString());
-  //     console.error("Error while connecting to Xverse wallet", err);
-  //   }
-  // };
+  const onConnectXverseWallet = async () => {
+    try {
+      const xverse = await ordit.xverse.getAddresses({
+        network,
+      });
+      // Nested Segwit = BTC
+      // Taproot = Ordinals / Inscriptions
+      const nestedSegwit = xverse.find((a) => a.format === "nested-segwit");
+      const taproot = xverse.find((a) => a.format === "taproot");
+      if (!nestedSegwit || !taproot) {
+        throw Error(
+          "Xverse extension is misbehaving. You may need to backup and restore your wallet.",
+        );
+      }
+      updateAddress({
+        ordinals: taproot.address,
+        payments: nestedSegwit.address,
+      });
+      updatePublicKey({ ordinals: taproot.pub, payments: nestedSegwit.pub });
+      updateWallet(Wallet.XVERSE);
+      updateFormat({
+        ordinals: taproot.format as AddressFormats,
+        payments: nestedSegwit.format as AddressFormats,
+      });
```

Comment:
before we connect, can we check that each of these from `getAddresses` exists if not throw error. Same for unisat side

Reply:
[I've only seen format being missing due to network related issues for Xverse, which is checked here.](https://github.com/sadoprotocol/sado-connect/blob/a8802d1f17d2895c4e23bcf63d20174536bc1d83/packages/sado-connect/src/components/SelectWalletModal/index.tsx#L83)

Reply:
hmm, I think it is not gonna be common but its good sanity check to ensure we throw error in the case that xverse breaks or breaking changes we can catch it as well 🤔